src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2704,26 +2703,22 @@ private BoundExpression BindImplicitArrayCreationExpression(ImplicitArrayCreatio
 
             bool hadMultipleCandidates;
             HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-            var bestType = BestTypeInferrer.InferBestType(
-                boundInitializerExpressions,
-                this.Conversions,
-                hadMultipleCandidates: out hadMultipleCandidates,
-                useSiteDiagnostics: ref useSiteDiagnostics);
+            TypeSymbol bestType = BestTypeInferrer.InferBestType(boundInitializerExpressions, this.Conversions, out hadMultipleCandidates, ref useSiteDiagnostics);
             diagnostics.Add(node, useSiteDiagnostics);
 
-            if (bestType.IsNull || bestType.SpecialType == SpecialType.System_Void) // Dev10 also reports ERR_ImplicitlyTypedArrayNoBestType for void.
+            if ((object)bestType == null || bestType.SpecialType == SpecialType.System_Void) // Dev10 also reports ERR_ImplicitlyTypedArrayNoBestType for void.
             {
                 Error(diagnostics, ErrorCode.ERR_ImplicitlyTypedArrayNoBestType, node);
-                bestType = TypeSymbolWithAnnotations.Create(CreateErrorType());
+                bestType = CreateErrorType();
             }
 
             if (bestType.IsRestrictedType())
             {
                 // CS0611: Array elements cannot be of type '{0}'
-                Error(diagnostics, ErrorCode.ERR_ArrayElementCantBeRefAny, node, bestType.TypeSymbol);
+                Error(diagnostics, ErrorCode.ERR_ArrayElementCantBeRefAny, node, bestType);
             }
 
-            var arrayType = ArrayTypeSymbol.CreateCSharpArray(Compilation.Assembly, bestType, rank);
+            var arrayType = ArrayTypeSymbol.CreateCSharpArray(Compilation.Assembly, TypeSymbolWithAnnotations.Create(bestType, isNullableIfReferenceType: null), rank);
```

Comment:
>isNullableIfReferenceType [](start = 127, length = 25)

Maybe comment that the nullability will be re-inferred based on flow information, so binding doesn't care.